### PR TITLE
Drop `serialized=False/readonly=True` for `NotSerialized/Readonly`

### PR DESCRIPTION
**Pull request by mattpap**
This is based on PR #12271, so only the last commit is relevant.

This PR removes `serialized` and `readonly` from properties' `__init__`, and replaces them with existing `Readonly()` and new `NotSerialized()`. Those `__init__` attributes only complicate the API, having to be passed through long inheritance chains, only to be used in `Property` base class. Due to this, many properties don't implement them at all or implement them incorrectly. However, they aren't being used anyway, especially since `Readonly()` was introduced. Thus my proposal is to drop them now. It should be a low impact breaking change.

### CODE REVIEW

File: bokeh/core/property/serialized.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    .. deprecated:: 3.0.0
+
+        Use ``bokeh.core.property.required.Required`` instead.
+    """
+
+    _serialized = False
+
+    def __init__(self, type_param: TypeOrInst[Property[T]], *, default: Init[T] = Intrinsic, help: str | None = None) -> None:
+        super().__init__(type_param, default=default, help=help)
```
Comment (bryevdv) [inline bokeh/core/property/serialized.py:60]: This should actually call `deprecated`

Comment (mattpap) [reply bokeh/core/property/serialized.py:60]: This is still a feature, which actually current test failures indicate (I removed one too many `serialized`).

Comment (bryevdv) [reply bokeh/core/property/serialized.py:60]: Sorry, I don't follow. The comment indicates it is deprecated, in which case a call to `deprecated` is advised to let users know that fact at runtime.

Comment (mattpap) [reply bokeh/core/property/serialized.py:60]: I didn't see enough context in to top-level thread, so I missed the deprecation. I copied this from `NonNullable. I will remove the deprecation.

**Pull request merged**